libraries/AP_RangeFinder/AP_RangeFinder_BLPing.cpp

```diff
@@ -149,49 +174,79 @@ void AP_RangeFinder_BLPing::send_message(uint16_t msgid, const uint8_t *payload,
     uart->write(HIGHBYTE(crc));
 }
 
-// distance returned in reading_cm, signal_ok is set to true if sensor reports a strong signal
-bool AP_RangeFinder_BLPing::get_reading(uint16_t &reading_cm)
+bool AP_RangeFinder_BLPing::check_new_messages()
 {
-    if (uart == nullptr) {
-        return false;
-    }
-
-    float sum_cm = 0;
-    uint16_t count = 0;
+    // check for at least a single valid message for the calling function
+    bool valid_message = false;
 
     // read any available lines from the lidar
     int16_t nbytes = uart->available();
+
+    // variables for arithmetic mean of distance_simples
+    uint16_t count = 0;
+    uint16_t mean_distance_cm = 0;
+    uint32_t sum_cm = 0;
+
     while (nbytes-- > 0) {
         const int16_t b = uart->read();
         if (b < 0) {
             break;
         }
-        if (parse_byte(b)) {
-            count++;
+        uint16_t message_id = parse_byte(b);
+
+        if(message_id == BLPING_MSGID_INVALID) {
+            continue;
+        }
+
+        // A single message is enough to set valid_message
+        valid_message = true;
+
+        // It's possible to receive multiple distance_simple messages
+        // The distance value is calculated with the arithmetic mean of n samples
+        if(BLDPIN_MSGID_DISTANCE_SIMPLE == message_id) {
             sum_cm += distance_cm;
+            count++;
+            mean_distance_cm = sum_cm/count;
```

Comment:
The average only needs to be calculated once after the accumulation, like randy had done.